File: solutions/c/resistor-color/2/resistor_color.c
#include "resistor_color.h"

unsigned int color_code(resistor_band_t color) {
    return (unsigned int) color;
}

static const resistor_band_t COLORS[10] = {
    0, 1, 2, 3, 4, 5, 6, 7, 8, 9
};

const resistor_band_t *colors() {
    return COLORS;
}

File: solutions/lua/resistor-color/2/resistor-color.lua
local COLORCODES = {
  black = 0,
  brown = 1,
  red = 2,
  orange = 3,
  yellow = 4,
  green = 5,
  blue = 6,
  violet = 7,
  grey = 8,
  white = 9,
}

return {
  color_code = function(color)
    if type(color) ~= "string" then
      error(string.format([[          
          color_code() expects a string input.
          received: %s
          ]], type(color)))
    end
    if COLORCODES[color] == nil then
      error(string.format([[          
          color_code() could not find the following color.
          color: %s
          ]], color))
    end
    return COLORCODES[color]
  end
}